stl/inc/xstring

```diff
@@ -2792,9 +2792,10 @@ public:
         _Container_proxy_ptr<_Alty> _Proxy(_Alproxy, _My_data); // throws
 
         if (_New_capacity < _New_size) {
-            _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size());
-            const pointer _Fancyptr = _Getal().allocate(_New_capacity + 1); // throws
+            _New_capacity           = _Calculate_growth(_New_size, _BUF_SIZE - 1, max_size()) + 1;
+            const pointer _Fancyptr = _Allocate_at_least_helper(_Getal(), _New_capacity); // throws
             _Ptr                    = _Unfancy(_Fancyptr);
+            _New_capacity--;

```

Comment:
Suggestion: It would seem simpler to delay this until the end of the `if` block (updating `_Start_element_lifetimes` accordingly) to make the code more consistent with the previous occurrences, and it make it clearer that we're "unifying" the interpretation of `_New_capacity` as we fall through to the code outside the condition.